detection-bot/src/agent.ts

```diff
@@ -0,0 +1,48 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
+
+import { sharedConstant, sharedConstantsType } from "./shared";
+/**
+ * Wrapper for the handleTransaction
+ * @param sharedConstant : Constants like createAgents function signature and to(polygon) and from(Nethermind) address.
+ * @returns findings: [] || array with alert object
+ */
+export function provideHandleTransaction(sharedConstant: sharedConstantsType): HandleTransaction {
+  return async function handleTransaction(txEvent: TransactionEvent): Promise<Finding[]> {
+    const findings: Finding[] = [];
+    if (
+      txEvent.from.toLowerCase() != sharedConstant.NETHERMIND_DEPLOYER_ADDRESS.toLowerCase() ||
+      txEvent.to?.toLowerCase() != sharedConstant.TO_REGISTER_POLYGON_ADDRESS.toLowerCase()
+    ) {
+      return findings;
+    }
+    const listAllRegisterTxEvent = txEvent.filterFunction(
+      sharedConstant.CREATE_AGENT_FUNCTION_SIGNATURE,
+      sharedConstant.TO_REGISTER_POLYGON_ADDRESS
+    );
+    listAllRegisterTxEvent.forEach((botDescription) => {
+      const { agentId, owner, metadata, chainIds } = botDescription.args;
+
+      findings.push(
+        Finding.fromObject({
+          name: "Nethermind Bot Detection",
```

Comment:
```suggestion
          name: "New Nethermind Bot Deployment Detection",
```

Suggesting a more descriptive name